dbus/examples/argument_guide.md

```diff
@@ -147,7 +147,69 @@ Structs
 D-Bus structs are implemented as Rust tuples. You can append and get tuples like you do with other types of arguments.
 You can also use `VecDeque<Box<dyn RefArg>>` for when the types of the struct cannot be statically typed.
 
-TODO: Example
+For an example of the latter, the [Desktop Notifications Specification](https://specifications.freedesktop.org/notification-spec/notification-spec-latest.html#idm46059356295632) specifies that the `notify()` method sends a dictionary of optional [hints](https://specifications.freedesktop.org/notification-spec/notification-spec-latest.html#hints), one of which is `"image-data"` hint, which contains a struct.  Here's how you might retrieve that `"image-data"` hint and cast it into a rust struct.
+
+Implementing the `notify()` method in Rust -- trait generated by [`dbus-codegen-rust`](https://github.com/diwic/dbus-rs/tree/master/dbus-codegen):
+```rust
+// impl OrgFreedesktopNotifications {
+fn notify(
+    &mut self,
+    app_name: String,
+    replaces_id: u32,
+    app_icon: String,
+    summary: String,
+    body: String,
+    actions: Vec<String>,
+    hints: arg::PropMap,
+    expire_timeout: i32
+) -> Result<u32, dbus::MethodErr> {
+    // PropMap is just an alias for a HashMap:
+    // pub type PropMap = HashMap<String, Variant<Box<dyn RefArg + 'static>>>;
+
+    use dbus::arg::prop_cast;
+
+    // Retrieving a property from a PropMap of type byte (`u8`).
+    let urgency: Option<&u8> = prop_cast(&hints, "urgency");
+    //let urgency: Option<u8> = prop_cast(&hints, "urgency").copied();
+    //let urgency = prop_cast::<u8>(&hints, "urgency");    // Option<&u8>
```

Comment:
These two comments could also be removed, not sure what they are for?

Reply:
Just wanted to show different ways of achieving the same thing.